tests/testthat/test-discrete_lognormal.R

```diff
@@ -0,0 +1,13 @@
+test_that(
+  "discrete_lognormal fails when given the wrong argument dimensions", {
+    # check breaks is a vector with at least length 2
+    expect_snapshot_error(
+      discrete_lognormal(p = 1, psi = 1, breaks = 1)
+    )
+    
+    # check dim to be a positive scalar integer
+    expect_snapshot_error(
+      discrete_lognormal(p = 1, psi = 1, breaks = c(1, 2), dim = 0.1)
```

Comment:
Shouldn't the parameters for `discrete_lognormal` be `meanlog` and `sdlog` ? Currently this will capture an error message about parameter names not being correct.

Reply:
Opps that was me copy-pasting mindlessly. But while I was at it, having a non-integer in the `dim` argument `discrete_lognormal(p = 1, psi = 1, breaks = c(1, 2), dim = 0.1)` did not error. It seems that non-integer `dim` will generate dimension that they are floored to, or whatever the value before the decimal takes. For example
```
> normal(0, 1, dim = 0.9)
greta array (variable following a normal distribution)

     [,1]
```
and 
```
> normal(0, 1, dim = 2.9)
greta array (variable following a normal distribution)

     [,1]
[1,]  ?  
[2,]  ? 
```
I think this mistake rarely happens day-to-day, so not sure if this should be checked in `greta` itself.

Reply:
Hmmm! We should check this, as dimension should strictly be an integer. One issue I can see this coming up is where people might programmatically change parameter values for generating arrays and perhaps the wrong parameter gets passed along.

We could write a little checker for integer values - what do you think @goldingn ?

Perhaps something like

``` r
0.1 %% 1
#> [1] 0.1
1.1 %% 1
#> [1] 0.1
1 %% 1
#> [1] 0
2 %% 1
#> [1] 0

check_is_integer <- function(x){
  is_int <- (x %% 1) == 0
  if (!is_int){
    msg <- cli::format_error(
      c(
        "{.val x} must be an integer",
        "However {.val x} is {.val {x}}"
      )
    )
    rlang::abort(
      msg
    )
  }
}

check_is_integer(1)
check_is_integer(1.1)
#> Error in `check_is_integer()`:
#> ! "x" must be an integer
#> However "x" is 1.1
```

<sup>Created on 2022-08-04 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

<details style="margin-bottom:10px;">
<summary>
Session info
</summary>

``` r
sessioninfo::session_info()
#> ─ Session info ───────────────────────────────────────────────────────────────
#>  setting  value
#>  version  R version 4.2.0 (2022-04-22)
#>  os       macOS Monterey 12.3.1
#>  system   aarch64, darwin20
#>  ui       X11
#>  language (EN)
#>  collate  en_AU.UTF-8
#>  ctype    en_AU.UTF-8
#>  tz       Australia/Perth
#>  date     2022-08-04
#>  pandoc   2.18 @ /Applications/RStudio.app/Contents/MacOS/quarto/bin/tools/ (via rmarkdown)
#> 
#> ─ Packages ───────────────────────────────────────────────────────────────────
#>  package     * version    date (UTC) lib source
#>  cli           3.3.0.9000 2022-06-15 [1] Github (r-lib/cli@31a5db5)
#>  crayon        1.5.1      2022-03-26 [1] CRAN (R 4.2.0)
#>  digest        0.6.29     2021-12-01 [1] CRAN (R 4.2.0)
#>  ellipsis      0.3.2      2021-04-29 [1] CRAN (R 4.2.0)
#>  evaluate      0.15       2022-02-18 [1] CRAN (R 4.2.0)
#>  fansi         1.0.3      2022-03-24 [1] CRAN (R 4.2.0)
#>  fastmap       1.1.0      2021-01-25 [1] CRAN (R 4.2.0)
#>  fs            1.5.2      2021-12-08 [1] CRAN (R 4.2.0)
#>  glue          1.6.2      2022-02-24 [1] CRAN (R 4.2.0)
#>  highr         0.9        2021-04-16 [1] CRAN (R 4.2.0)
#>  htmltools     0.5.2      2021-08-25 [1] CRAN (R 4.2.0)
#>  knitr         1.39       2022-04-26 [1] CRAN (R 4.2.0)
#>  lifecycle     1.0.1      2021-09-24 [1] CRAN (R 4.2.0)
#>  magrittr      2.0.3      2022-03-30 [1] CRAN (R 4.2.0)
#>  pillar        1.7.0      2022-02-01 [1] CRAN (R 4.2.0)
#>  pkgconfig     2.0.3      2019-09-22 [1] CRAN (R 4.2.0)
#>  purrr         0.3.4      2020-04-17 [1] CRAN (R 4.2.0)
#>  R.cache       0.15.0     2021-04-30 [1] CRAN (R 4.2.0)
#>  R.methodsS3   1.8.1      2020-08-26 [1] CRAN (R 4.2.0)
#>  R.oo          1.24.0     2020-08-26 [1] CRAN (R 4.2.0)
#>  R.utils       2.11.0     2021-09-26 [1] CRAN (R 4.2.0)
#>  reprex        2.0.1      2021-08-05 [1] CRAN (R 4.2.0)
#>  rlang         1.0.4      2022-07-12 [1] CRAN (R 4.2.0)
#>  rmarkdown     2.14       2022-04-25 [1] CRAN (R 4.2.0)
#>  rstudioapi    0.13       2020-11-12 [1] CRAN (R 4.2.0)
#>  sessioninfo   1.2.2      2021-12-06 [1] CRAN (R 4.2.0)
#>  stringi       1.7.6      2021-11-29 [1] CRAN (R 4.2.0)
#>  stringr       1.4.0      2019-02-10 [1] CRAN (R 4.2.0)
#>  styler        1.7.0      2022-03-13 [1] CRAN (R 4.2.0)
#>  tibble        3.1.7      2022-05-03 [1] CRAN (R 4.2.0)
#>  utf8          1.2.2      2021-07-24 [1] CRAN (R 4.2.0)
#>  vctrs         0.4.1      2022-04-13 [1] CRAN (R 4.2.0)
#>  withr         2.5.0      2022-03-03 [1] CRAN (R 4.2.0)
#>  xfun          0.31       2022-05-10 [1] CRAN (R 4.2.0)
#>  yaml          2.3.5      2022-02-21 [1] CRAN (R 4.2.0)
#> 
#>  [1] /Library/Frameworks/R.framework/Versions/4.2-arm64/Resources/library
#> 
#> ──────────────────────────────────────────────────────────────────────────────
```

</details>